turbinia/api/api_server.py

```diff
@@ -0,0 +1,153 @@
+          'name': 'Apache License 2.0',
+          'url': 'https://www.apache.org/licenses/LICENSE-2.0.html'
+      }, routes=router.routes)
+  return _app
+
+
+def set_operation_ids(app):
+  """Simplify operation IDs so that generated API clients have
+    simpler function names.
+    Should be called only after all routes have been added.
+    """
+  for route in app.routes:
+    if isinstance(route, APIRoute):
+      route.operation_id = route.name
+
+
+def serve_static_content(app):
+  """Configure the application to serve static content.
+
+  This method must be called after all routes have been initialized.
+  """
+  root_content_path = os.path.realpath('../../web')
+  css_content_path = os.path.join(root_content_path, 'css')
+  js_content_path = os.path.join(root_content_path, 'js')
+
+  app.mount("/", StaticFiles(directory=root_content_path, html=True), name="/")
+  app.mount("/css", StaticFiles(directory=css_content_path), name="/css")
+  app.mount("/js", StaticFiles(directory=js_content_path), name="/js")
+
+
+def configure_authentication_providers(app):
+  """Configure the application's authentication providers.
+
+  Example provider configuration using starlette-oauth2-pai:
+
+  app.add_middleware(
+      AuthenticateMiddleware,
+      providers={
+          'google-ui': {
+              'keys':
+                  'https://www.googleapis.com/oauth2/v3/certs',
+              'issuer':
+                  'https://accounts.google.com',
+              'audience':
+                  'GOOGLE_WEB_CLIENT_ID',
+          },
+          'cli-client': {
+              'keys':
+                  'https://www.googleapis.com/oauth2/v3/certs',
+              'issuer':
+                  'https://accounts.google.com',
+              'audience':
+                  'GOOGLE_NATIVE_CLIENT_ID',
+          }
+      },
+      public_paths={'/'},
+  )
+  """
+  raise NotImplementedError
+
+
+app = get_application()
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=config.API_ALLOWED_ORIGINS,
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+if config.API_AUTHENTICATION_ENABLED:
+  configure_authentication_providers(app)
+
+set_operation_ids(app)
+#serve_static_content(app)
+
+
+class TurbiniaAPIServer:
+  """Turbinia API server."""
+
+  def __init__(self, app=None, router=None):
+    self.app = app if app else get_application()
+    self.router = router
+    self.openapi_spec = self.app.openapi()
+
+  def start(self, app_name: str):
+    """Runs the Turbinia API server
+
+    Args:
+      app_name (str): module:app string used by Uvicorn
+          to start the HTTP server.
+    """
+    _config = config.LoadConfig()
+    uvicorn.run(
+        app_name, host=_config.API_SERVER_ADDRESS, port=_config.API_SERVER_PORT,
+        log_level="info", reload=True)
+
+
+@app.get('/docs/openapi.yaml', include_in_schema=False)
+def read_openapi_yaml():
+  """Serve the OpenAPI specification in YAML format."""
```

Comment:
Do we use YAML format to generate the client as the client tools do not support JSON?